src/template_helpers.rs

```diff
@@ -251,25 +252,76 @@ fn typeof_helper(v: &JsonValue) -> JsonValue {
     .into()
 }
 
+pub trait MarkdownConfig {
+    fn allow_dangerous_html(&self) -> bool;
+    fn allow_dangerous_protocol(&self) -> bool;
+}
+
+impl MarkdownConfig for AppConfig {
+    fn allow_dangerous_html(&self) -> bool {
+        self.markdown_allow_dangerous_html
+    }
+
+    fn allow_dangerous_protocol(&self) -> bool {
+        self.markdown_allow_dangerous_protocol
+    }
+}
+
 /// Helper to render markdown with configurable options
 struct MarkdownHelper {
     allow_dangerous_html: bool,
     allow_dangerous_protocol: bool,
 }
 
+impl Default for MarkdownHelper {
+    fn default() -> Self {
+        Self {
+            allow_dangerous_html: false,
+            allow_dangerous_protocol: false,
+        }
+    }
+}
+
 impl MarkdownHelper {
-    fn new(config: &AppConfig) -> Self {
+
+    const ALLOW_UNSAFE: &'static str = "allow_unsafe";
+
+    fn new(config: &impl MarkdownConfig) -> Self {
         Self {
-            allow_dangerous_html: config.markdown_allow_dangerous_html,
-            allow_dangerous_protocol: config.markdown_allow_dangerous_protocol,
+            allow_dangerous_html: config.allow_dangerous_html(),
+            allow_dangerous_protocol: config.allow_dangerous_protocol(),
         }
     }
+
+    fn calculate_options(&self, args: &[PathAndJson]) -> Options {
+        let mut options = self.system_options();
+
+        if !options.compile.allow_dangerous_html && args.len() > 1 {
+            let arg = &args[1];
+            if arg.relative_path() == Some(&Self::ALLOW_UNSAFE.to_string()) {
```

Comment:
It's strange to use the path instead of the value. This means you cannot pass a variable to your helper. We should probably read the string value of the argument here.

Reply:
This was based on my logging of what was going on with different combinations passed into the handler; I'll switch it to use the variable.